tests/tests/mempool.rs

```diff
@@ -351,3 +367,105 @@ pub async fn create_sample_transactions(
+    //
+    // This task will periodically prune the mempool based on the given prune_duration.
+    // For testing purposes, we set the prune_duration to 1 second.
+    let prune_duration = Duration::from_secs(1);
+    let eth_client_clone = Arc::clone(&eth_client);
+    let maintain_task = tokio::spawn(async move {
+        maintain_transaction_pool(eth_client_clone, prune_duration);
+    });
+
+    // Initialize the block number based on the current blockchain state from katana.
+    let mut last_block_number = katana.block_number();
+
+    // Loop to simulate new blocks being added to the blockchain every 100 milliseconds.
+    for _ in 0..10 {
+        // Sleep for 100 milliseconds to simulate the passage of time between blocks.
+        tokio::time::sleep(Duration::from_millis(100)).await;
+
+        // Increment the block number to simulate the blockchain progressing.
+        last_block_number += 1;
+
+        // Format the block number in both padded and unpadded hexadecimal formats.
+        let unpadded_block_number = format_hex(last_block_number, 0);
+        let padded_block_number = format_hex(last_block_number, U64_HEX_STRING_LEN);
+
+        // Get the block header collection from the database.
+        let header_collection = eth_client.eth_provider().database().collection::<StoredHeader>();
+
+        // Build a filter for updating the header based on the new block number.
+        let filter = EthDatabaseFilterBuilder::<filter::Header>::default().with_block_number(last_block_number).build();
+
+        // Insert a new header for the new block number in the database.
+        eth_client
+            .eth_provider()
+            .database()
+            .update_one(
+                StoredHeader {
+                    header: Header {
+                        hash: B256::random(),
+                        total_difficulty: Some(U256::default()),
+                        mix_hash: Some(B256::default()),
+                        nonce: Some(B64::default()),
+                        withdrawals_root: Some(EMPTY_ROOT_HASH),
+                        base_fee_per_gas: Some(0),
+                        blob_gas_used: Some(0),
+                        excess_blob_gas: Some(0),
+                        number: last_block_number,
+                        ..Default::default()
+                    },
+                },
+                filter,
+                true,
+            )
+            .await
+            .expect("Failed to update header in database");
+
+        // Update the header collection with the padded block number in the database.
+        header_collection
+            .update_one(
+                doc! {"header.number": unpadded_block_number},
+                UpdateModifications::Document(doc! {"$set": {"header.number": padded_block_number}}),
+            )
+            .with_options(UpdateOptions::builder().upsert(true).build())
+            .await
+            .expect("Failed to update block number");
```

Comment:
We need to scope an issue here to be able to add block headers for testing more easily in the database. This behavior is also present in katana. As it is for testing purposes, we do not have a dedicated function in the production flow but we would have to scope an issue to do this properly.

Reply:
Can we create an issue and scope it in next tech workshop?

Reply:
done https://github.com/kkrt-labs/kakarot-rpc/issues/1489